hydromt_sfincs/utils.py

```diff
@@ -876,37 +881,133 @@ def downscale_floodmap(
     if timedim:
         zsmax = zsmax.max(timedim)
 
-    # interpolate zsmax to dep grid
-    zsmax = zsmax.raster.reproject_like(dep, method=reproj_method)
-    zsmax = zsmax.raster.mask_nodata()  # make sure nodata is nan
-
-    # get flood depth
-    hmax = (zsmax - dep).astype("float32")
-    hmax.raster.set_nodata(np.nan)
-
-    # mask floodmap
-    hmax = hmax.where(hmax > hmin)
-    if gdf_mask is not None:
-        mask = hmax.raster.geometry_mask(gdf_mask, all_touched=True)
-        hmax = hmax.where(mask)
+    if isinstance(dep, xr.DataArray):
+        hmax = _downscale_floodmap_da(
+            zsmax=zsmax,
+            dep=dep,
+            hmin=hmin,
+            gdf_mask=gdf_mask,
+            reproj_method=reproj_method,
+        )
 
-    # write floodmap
-    if floodmap_fn is not None:
-        if not kwargs:  # write COG by default
-            kwargs = dict(
+        # write floodmap
+        if floodmap_fn is not None:
+            if not kwargs:  # write COG by default
+                kwargs = dict(
+                    driver="GTiff",
+                    tiled=True,
+                    blockxsize=256,
+                    blockysize=256,
+                    compress="deflate",
+                    predictor=2,
+                    profile="COG",
+                )
+            hmax.raster.to_raster(floodmap_fn, **kwargs)
+
+            # add overviews
+            build_overviews(fn=floodmap_fn, resample_method="nearest")
+
+        hmax.name = "hmax"
+        hmax.attrs.update({"long_name": "Maximum flood depth", "units": "m"})
+        return hmax
+
+    elif isinstance(dep, (str, Path)):
+        assert (
+            floodmap_fn is not None
+        ), "floodmap_fn should be provided when dep is a Path or str."
```

Comment:
ValueError (or more specific errors) are considered better practice than assertions in python packages. arrestion errors are meant for testing / debuging.

```suggestion
        if floodmap_fn is not None:
            raise ValueError("floodmap_fn should be provided when dep is a Path or str.")
```